### DISCUSSION TITLE
PlatformaIO

### DISCUSSION THREAD

**Discussion by VaBanck** [rednblkx/HomeKey-ESP32#51] [Q&A] [answered]
What could be the problem? When compiling, it gives such an error![image](https://github.com/rednblkx/HomeKey-ESP32/assets/75133860/0ba87a63-8def-4a9b-9161-e3d077a3b02f)![image](https://github.com/rednblkx/HomeKey-ESP32/assets/75133860/b4e9baeb-277d-4b87-b05c-983bbd551c48)

**Comment by rednblkx** [comment]
Not errors, unless they are for some reason preventing you from compiling, they are just warnings as also clearly denoted.

**Reply by VaBanck** [reply]
https://github.com/rednblkx/HomeKey-ESP32/assets/75133860/53571053-f818-4c36-b97d-78839425445c

Works without setting up mqtt (only connection to mqtt broker) is it possible to do that it closed the lock?

**Reply by rednblkx** [reply]
what do you mean "closed"?

**Reply by VaBanck** [reply]
The video shows that I'm opening the lock, I can't close it. Sorry for my English

**Reply by ihrapsa** [reply]
Afaik the dummy lock entity needs to be always controlled by mqtt. In other words, you need to have a working automation that catches the mqtt state messages and respond accordingly.

**Reply by rednblkx** [reply, marked as answer]
basically what @ihrapsa said.

But i want to explain a couple of things, since it does change to unlock and not lock, i assume you enabled `MQTT_HOMEKEY_ALWAYS_UNLOCK` as normally it doesn't do anything, this option makes it so it publishes the value 0 for UNLOCK to the `MQTT_STATE_TOPIC` mqtt topic and also updates the homekit entity to Unlocked.

In case you tried to enable both `MQTT_HOMEKEY_ALWAYS_UNLOCK` and `MQTT_HOMEKEY_ALWAYS_LOCK`, note that's not possible since that wouldn't make sense and also the code checks with an "if else" if `MQTT_HOMEKEY_ALWAYS_UNLOCK` is enable else if `MQTT_HOMEKEY_ALWAYS_LOCK` is enabled.
